src/coreclr/jit/inlinepolicy.cpp

```diff
@@ -727,6 +735,22 @@ double DefaultPolicy::DetermineMultiplier()
         JITDUMP("\nInline candidate has arg that feeds range check.  Multiplier increased to %g.", multiplier);
     }
 
+    if (m_ConstArgFeedsIsKnownConst)
+    {
+        // if we use RuntimeHelpers.IsKnownConstant we most likely expect our function to be always inlined
+        // at least in the case of constant arguments
+        multiplier += 15;
+        JITDUMP("\nConstant argument feeds RuntimeHelpers.IsKnownConstant.  Multiplier increased to %g.", multiplier);
+    }
+
+    if (m_ArgFeedsIsKnownConst)
+    {
+        // In IsPrejitRoot we don't have callsite info so let's assume we have a constant here in order to avoid "baked"
```

Comment:
For my education, what is the callsite info that is missing for IsPrejitRoot?

Reply:
@jkotas It's a separate run during prejitting where we just take every function (without context) and try to memorize all those which will never be inlined (by reporting to VM "mark this function as noinline" [here](https://github.com/dotnet/runtime/blob/c912a1532a1bb653a21a3749aaad7c5fb556ae01/src/coreclr/jit/inline.cpp#L794-L796)). During that, we opportunistically assume the context is the most inliner friendly: all imaginary arguments are constants, imaginary callsite has PGO data and is hot, etc.. there was a better/detailed explanation from Andy somewhere

Reply:
Right, we leverage prejitting to mark methods that will never be inlined so we don't spend any jit time analyzing them.